### PR TITLE
Drop support for Swift 5.2 and 5.3

### PR DESCRIPTION
**Pull request by fabianfett**
As outlined in a [Swift forums post in November ’21](https://forums.swift.org/t/swiftnio-swift-version-support/53232), SwiftNIO will only support the latest non-patch Swift release and the 2 immediately prior non-patch versions.

In this commit we drop support for Swift 5.2 and 5.3. We update CI for Swift 5.4 to run on bionic instead of focal to ensure that we still test bionic.

### CODE REVIEW

**Review by Lukasa — APPROVED** [review]
Subject to CI changes getting applied

**Comment by fabianfett** [comment]
@yim-lee Could you please disable the CI for 5.2 and 5.3 and change the CI file for 5.4 from `docker/docker-compose.2004.54.yaml` to `docker/docker-compose.1804.54.yaml`.

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Repository | NIO 2 (Swift 5.2+)
 [https://github.com/apple/swift-nio-transport-services][repo-nio-transport-services] <br> first-class support for macOS, iOS, tvOS, and watchOS | `from: "1.0.0"`
 [https://github.com/apple/swift-nio-ssh][repo-nio-ssh] <br> SSH support | `.upToNextMinor(from: "0.2.0")`
 
-NIO 2.29.0 and older support Swift 5.0+.
+NIO 2.29.0 and older support Swift 5.0+, NIO 2.39.0 and older support Swift 5.2+.
```
Comment (weissi) [inline README.md:23]: All happy with that. Mind adding some official wording regarding security updates. Let's assume a security vulnerability in NIO 2.29.0 and newer gets identified. Who will get patches? I assume that at this point it'll be three updates: One for 2.29.x, one for 2.39.x and one for latest. Correct?

Comment (Lukasa) [reply README.md:23]: Assuming the vulnerability exists in all three, I think that's the policy, yes.

Comment (weissi) [reply README.md:23]: Sounds good. I think this should just be stated explicitly. It's clear (from elsewhere in that doc) that NIO 1.x won't get security updates anymore (as of May 2022) but that doesn't make it explicit that NIO 2.x _will_ get them for the time being.

**Comment by yim-lee** [comment]
I disabled CI 5.2 and 5.3, but since there are in-progress/pending PRBs I haven't updated 5.4 yet. Should I go ahead?

**Comment by Lukasa** [comment]
I think it's fine to update 5.4: those PRs can queue behind this one and we can rebase them to get the tests to run.

**Comment by yim-lee** [comment]
5.4 updated to use 18.04 instead. 

@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

File: SECURITY.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+
+* NIO 2.29. + plus next patch release to address the issue for projects that support
+  Swift 5.0 and 5.1
+* NIO 2.39. + plus next patch release to address the issue for projects that support 
+  Swift 5.2 and 5.3
+* mainline + plus next patch release to address the issue for projects that support 
+  Swift 5.4 and later
+
+SwiftNIO 1.x is considered end of life and will not receive any security patches.
+
```
Comment (fabianfett) [inline SECURITY.md:22]: @weissi @Lukasa Added a security version paragraph.

File: SECURITY.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
 
 This document specifies the security process for the SwiftNIO project.
 
+## Versions
+
+The SwiftNIO core team will address Security vulnarabilities in all SwiftNIO 2.x
```
Comment (Lukasa) [inline SECURITY.md:7]:
```suggestion
The SwiftNIO core team will address security vulnerabilities in all SwiftNIO 2.x 
```

File: SECURITY.md [hunk 4]
```diff
@@ -2,6 +2,24 @@
 
 This document specifies the security process for the SwiftNIO project.
 
+## Versions
+
+The SwiftNIO core team will address Security vulnarabilities in all SwiftNIO 2.x 
+versions. Since support for some Swift versions was dropped during the lifetime of 
+SwiftNIO 2, patch releases will be created for the last supported SwiftNIO versions
+that supported older Swift versions. 
+Considering a security vulnerability was introduced in 2.10.0, then SwiftNIO core
```
Comment (Lukasa) [inline SECURITY.md:11]:
```suggestion
If a hypothetical security vulnerability was introduced in 2.10.0, then SwiftNIO core 
```

File: SECURITY.md [hunk 5]
```diff
@@ -2,6 +2,24 @@
...
 This document specifies the security process for the SwiftNIO project.
 
+## Versions
+
+The SwiftNIO core team will address Security vulnarabilities in all SwiftNIO 2.x 
+versions. Since support for some Swift versions was dropped during the lifetime of 
+SwiftNIO 2, patch releases will be created for the last supported SwiftNIO versions
+that supported older Swift versions. 
+Considering a security vulnerability was introduced in 2.10.0, then SwiftNIO core 
+team will create the following patch releases:
```
Comment (Lukasa) [inline SECURITY.md:12]:
```suggestion
team would create the following patch releases:
```

**Review by Lukasa — APPROVED** [review]

**Pull request merged**